Player.cpp

```diff
@@ -0,0 +1,54 @@
+#include "Player.h"
+#include "Entity.h"
+#include <string>
+#include <iostream>
+#include <cmath>
+
+Player::Player(const int health,const int attackDamage,const std::string& name) : ExperienceCurrent(0), Entity(health, attackDamage, name) {}
+
+Player::Player(const Player& other) : Entity(other.Health, other.AttackDamage, other.Name){
+    this->MaxHealth                     =   other.MaxHealth;
+    this->Health                        =   other.Health;
+    this->AttackDamage                  =   other.AttackDamage;
+    this->Name                          =   other.Name;
+    this->ExperienceCurrent             =   other.ExperienceCurrent;
+}
+
+Player& Player::operator=(const Player &player){
+    this->MaxHealth                     =   player.MaxHealth;
+    this->Health                        =   player.Health;
+    this->AttackDamage                  =   player.AttackDamage;
+    this->Name                          =   player.Name;
```

Comment:
Itt is lehetne ezekhez használni az `Entity::operator=`-t. Meg egy `dynamic_cast`tal össze lehetne vonni ezt a két assignment operátort.